File: src/main/scala/com/stulsoft/swing/exercises/panels/BorderLayoutPanel.scala
/*
 * Copyright (c) 2023. StulSoft
 */

package com.stulsoft.swing.exercises.panels

import java.awt.Color
import scala.swing.*
import scala.swing.BorderPanel.Position
import scala.swing.Component.*

object BorderLayoutPanel extends SimpleSwingApplication:
  override def top: Frame = new MainFrame {
    title = "Border layout panel"
    val l = new Label("panel label (Center)")

    val b = new Button("North")
    val b1 = new Button("South")
    val b2 = new Button("East")
    val b3 = new Button("West")

    contents = new BorderPanel {
      layout(b) = Position.North
      layout(b1) = Position.South
      layout(b2) = Position.East
      layout(b3) = Position.West
      layout(l) = Position.Center
      background = Color.red
    }

    size = new Dimension(400, 600)
    centerOnScreen()
  }

File: src/main/java/com/stulsoft/swing/jexamples/panels/BorderLayoutPanel.java
/*
 * Copyright (c) 2023. StulSoft
 */

package com.stulsoft.swing.jexamples.panels;

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class BorderLayoutPanel extends JFrame {
    // JFrame
    static JFrame f;

    // JButton
    static JButton b, b1, b2, b3;

    // Label to display text
    static JLabel l;

    // Main driver method
    public static void main(String[] args) {
        // Creating a new frame to store text field and
        // button
        f = new JFrame("panel");

        // Creating a label to display text
        l = new JLabel("panel label");

        // Creating a new buttons
        b = new JButton("button1");
        b1 = new JButton("button2");
        b2 = new JButton("button3");
        b3 = new JButton("button4");

        // Creating a panel to add buttons
        // and a specific layout
        JPanel p = new JPanel(new BorderLayout());

        // Adding buttons and textfield to panel
        // using add() method
        p.add(b, BorderLayout.NORTH);
        p.add(b1, BorderLayout.SOUTH);
        p.add(b2, BorderLayout.EAST);
        p.add(b3, BorderLayout.WEST);
        p.add(l, BorderLayout.CENTER);

        // setbackground of panel
        p.setBackground(Color.red);

        // Adding panel to frame
        f.add(p);

        // Setting the size of frame
        f.setSize(300, 300);
        f.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
        f.show();
    }
}
